Sources/SwiftKueryORM/Model.swift

```diff
@@ -341,55 +341,83 @@ public extension Model {
 
   /// - Parameter matching: Optional QueryParams to use
   /// - Returns: An array of model
-  static func findAll<Q: QueryParams>(using db: Database? = nil, matching queryParams: Q, _ onCompletion: @escaping ([Self]?, RequestError?) -> Void) {
+  static func findAll<Q: QueryParams>(using db: Database? = nil, matching queryParams: Q? = nil, _ onCompletion: @escaping ([Self]?, RequestError?) -> Void) {
     var table: Table
-    var filter: Filter
     do {
       table = try Self.getTable()
-      filter = try Self.getFilter(queryParams: queryParams, table: table)
     } catch let error {
       onCompletion(nil, Self.convertError(error))
       return
     }
 
-    let query = Select(from: table).where(filter)
-    Self.executeQuery(query: query, using: db, onCompletion)
+    var query: Select = Select(from: table)
+    var parameters: [Any?]? = nil
+    if let queryParams = queryParams {
+      do {
+        let resultTuple: (query: Select, parameters: [Any?]?) = try getSelectQueryWithFilters(query: query, queryParams: queryParams, table: table)
```

Comment:
I don't think you need the intermediate tuple, you can just do:

```swift
(query, parameters) = try getSelectQueryWithFilters(query: query, queryParams: queryParams, table: table)
```

Reply:
Done, that is much nicer!